sklearn/model_selection/_search.py

```diff
@@ -320,24 +324,61 @@ def fit_grid_point(X, y, estimator, parameters, train, test, scorer,
     return score, parameters, n_samples_test
 
 
+def _check_param_dist(param_dist):
```

Comment:
Indeed, the check for `isinstance(parameter_iterable, Sized)` suggests that someone wanted support even when the iterable has no `len`.